### PR TITLE
added current limit to motor

### PR DESCRIPTION
**Pull request by christiancruzfhs**

### CODE REVIEW

File: Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 1]
```diff
@@ -64,12 +67,14 @@ public ElevatorSubsystem(CommonLibFactory clf, XPropertyManager propMan, Electri
...
         currentHeight = propMan.createEphemeralProperty("Elevator current height", 0.0);
         lowerLimitSensor = propMan.createEphemeralProperty("Elevator Lower Limit", false);
+        upperLimitSensor = propMan.createEphemeralProperty("Elevator Upper Limit", false);
         targetScaleHighHeight = propMan.createPersistentProperty("Elevator scale high", 76.5);
         targetScaleMidHeight = propMan.createPersistentProperty("Elevator scale mid", 64.5);
         targetSwitchDropHeight = propMan.createPersistentProperty("Elevator switch drop height", 19.0);
         targetPickUpHeight = propMan.createPersistentProperty("Elevator pickup height", 3.0);
-
-        upperLimitSensor = propMan.createEphemeralProperty("Elevator Upper Limit", false);
+        peakCurrentLimit = propMan.createPersistentProperty("Peak current limit", 35);
```
Comment (JohnGilb) [inline Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: `Elevator Peak Current Limit`

File: Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 2]
```diff
@@ -64,12 +67,14 @@ public ElevatorSubsystem(CommonLibFactory clf, XPropertyManager propMan, Electri
...
+        peakCurrentDuration = propMan.createPersistentProperty("Peak current duration", 200);
```
Comment (JohnGilb) [inline Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: etc

File: Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 3]
```diff
@@ -64,12 +67,14 @@ public ElevatorSubsystem(CommonLibFactory clf, XPropertyManager propMan, Electri
...
+        continuousCurrentLimit = propMan.createPersistentProperty("Continuous current limit", 30);
```
Comment (JohnGilb) [inline Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: etc

File: Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 4]
```diff
@@ -99,6 +104,11 @@ private void initializeMotor() {
         motor.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
         motor.setSensorPhase(true);
 
+        motor.configPeakCurrentLimit((int) peakCurrentLimit.get(), 0);
```
Comment (JohnGilb) [inline Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: choose defaults that aren't 0, or this robot will never move.

**Review by JohnGilb — CHANGES_REQUESTED** [review]

**Review by JohnGilb — APPROVED** [review]

File: Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 5]
```diff
@@ -53,6 +53,10 @@
     private final DoubleProperty targetScaleMidHeight;
     private final DoubleProperty targetSwitchDropHeight;
     private final DoubleProperty targetPickUpHeight;
+    final BooleanProperty upperLimitSensor;
```
Comment (WasabiFan) [inline Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: `private`

File: Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 6]
```diff
@@ -108,6 +115,11 @@ private void initializeMotor() {
         motor.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
         motor.setSensorPhase(true);
 
+        motor.configPeakCurrentLimit((int) elevatorPeakCurrentLimit.get(), 0);
```
Comment (WasabiFan) [inline Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:119]: Property changes will never be propagated to the Talon. Do we want to do something like @JohnGilb did for the drive?

File: Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 7]
```diff
@@ -108,6 +115,11 @@ private void initializeMotor() {
...
+        motor.configPeakCurrentDuration((int) elevatorPeakCurrentDuration.get(), 0);
+        motor.configContinuousCurrentLimit((int) elevatorContinuousCurrentLimit.get(), 0);
+        motor.enableCurrentLimit(true);
```
Comment (WasabiFan) [inline Competition2018/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:122]: Property/command/etc. to disable current limit if it causes trouble?

**Review by JohnGilb — CHANGES_REQUESTED** [review]
Do what Kaelin says =]

**Review by JohnGilb — APPROVED** [review]

**Review by JohnGilb — APPROVED** [review]

**Pull request merged**